code/modules/reagents/newchem/medicine.dm

```diff
@@ -0,0 +1,711 @@
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	overdose_threshold = 25
+
+datum/reagent/sal_acid/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	if(M.getBruteLoss() < 50)
+		if(prob(50))
+			M.adjustBruteLoss(-1*REM)
+	..()
+	return
+
+datum/reagent/sal_acid/overdose_process(var/mob/living/M as mob)
+	if(M.getBruteLoss() < 50)
+		if(prob(50))
+			M.adjustBruteLoss(2*REM)
+	..()
+	return
+
+/datum/chemical_reaction/sal_acid
+	name = "Salicyclic Acid"
+	id = "sal_acid"
+	result = "sal_acid"
+	required_reagents = list("sodium" = 1, "phenol" = 1, "carbon" = 1, "oxygen" = 1, "sacid" = 1)
+	result_amount = 5
+
+datum/reagent/salbutamol
+	name = "Salbutamol"
+	id = "salbutamol"
+	description = "Quickly heals oxygen damage while slowing down suffocation. Great for stabilizing critical patients!"
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.2
+
+datum/reagent/salbutamol/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.adjustOxyLoss(-6*REM)
+	if(M.losebreath >= 4)
+		M.losebreath -= 4
+	..()
+	return
+
+/datum/chemical_reaction/salbutamol
+	name = "Salbutamol"
+	id = "salbutamol"
+	result = "salbutamol"
+	required_reagents = list("sal_acid" = 1, "lithium" = 1, "aluminium" = 1, "bromine" = 1, "ammonia" = 1)
+	result_amount = 5
+
+datum/reagent/perfluorodecalin
+	name = "Perfluorodecalin"
+	id = "perfluorodecalin"
+	description = "Heals suffocation damage so quickly that you could have a spacewalk, but it mutes your voice. Has a 33% chance of healing brute and burn damage per cycle as well."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.2
+
+datum/reagent/perfluorodecalin/on_mob_life(var/mob/living/carbon/human/M as mob)
+	if(!M) M = holder.my_atom
+	M.adjustOxyLoss(-25*REM)
+	M.silent = max(M.silent, 5)
+	if(prob(33))
+		M.adjustBruteLoss(-1*REM)
+		M.adjustFireLoss(-1*REM)
+	..()
+	return
+
+/datum/chemical_reaction/perfluorodecalin
+	name = "Perfluorodecalin"
+	id = "perfluorodecalin"
+	result = "perfluorodecalin"
+	required_reagents = list("hydrogen" = 1, "fluorine" = 1, "oil" = 1)
+	result_amount = 3
+	required_temp = 370
+	mix_message = "The mixture rapidly turns into a dense pink liquid."
+
+datum/reagent/ephedrine
+	name = "Ephedrine"
+	id = "ephedrine"
+	description = "Reduces stun times, increases run speed. If overdosed it will deal toxin and oxyloss damage."
+	reagent_state = LIQUID
+	color = "#C8A5DC"
+	metabolization_rate = 0.3
+	overdose_threshold = 45
+	addiction_threshold = 30
+	effect_speed = -1
+
+datum/reagent/ephedrine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	M.AdjustParalysis(-1)
+	M.AdjustStunned(-1)
+	M.AdjustWeakened(-1)
+	..()
+	return
+
+datum/reagent/ephedrine/overdose_process(var/mob/living/M as mob)
+	if(prob(33))
+		M.adjustToxLoss(1*REM)
+		M.losebreath++
+	..()
+	return
+
+datum/reagent/ephedrine/addiction_act_stage1(var/mob/living/M as mob)
```

Comment:
Rather than making these all separate procs <s>god, why</s>
Make it a singe proc and pass a stage var.